includes/utilities.inc

```diff
@@ -426,6 +426,53 @@ function islandora_paged_content_pdf_combine(array $files, $out) {
 }
 
 /**
+ * Appends a series of OCR files to a consolidated OCR file.
+ *
+ * @param string $consolidated_ocr
+ *   The consolidated OCR path.
+ * @param array $files
+ *   The files to be appended to the OCR.
+ *
+ * @return bool
+ *   TRUE on success, FALSE on failure.
+ */
+function islandora_paged_content_ocr_append($consolidated_ocr, array $files) {
+  $temp_file = "$consolidated_ocr.temp.txt";
+  copy($file, $temp_file);
+  array_unshift($files, $temp_file);
+  $ret = islandora_paged_content_ocr_combine($files, $file);
+  file_unmanaged_delete($temp_file);
+  return $ret;
+}
+
+/**
+ * Combines the given OCR files into one output file.
+ *
+ * @param array $files
+ *   The OCR files to be combined, in order.
+ * @param string $out
+ *   The absolute path to the consolidated OCR file.
+ *
+ * @return bool
+ *   TRUE on success, FALSE on fail.
+ */
+function islandora_paged_content_ocr_combine(array $files, $out) {
+  $success = TRUE;
+  $combined_ocr = '';
+  foreach ($files as $file) {
+    $ocr = file_get_contents($file);
+    if ($ocr == FALSE) {
+      $success = FALSE;
+    }
+    else {
+      $combined_ocr .= "$ocr\n\n";
+    }
+  }
+  $result = file_put_contents($out, rtrim($combined_ocr), FILE_APPEND);
```

Comment:
If we're worried about memory usage, we may be better off `fopen()`ing and writing each entry individually, or just using `file_put_contents()` for the contents of each file... Seems inconsistent to concatenate a set in memory and then dump the big blob to a file, when we could just dump to the file. Using `fopen()` may be easier on buffers (compared to multiple `file_put_contents()` calls... Shrug?